modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/PageRenderer/Page.es.js

```diff
@@ -177,6 +179,53 @@ const Page = ({
 		DDM_FORM_PORTLET_NAMESPACE
 	);
 
+	const {defaultLanguageId, history} = useFormState();
```

Comment:
why did you move the `useEffect` from the `WebContentVariant`? just want to know the reason behind it, since on `WebContentVariant` was more safe and clean since it only affects the web content page, while the Page file corresponds to all the features that use data engine

Reply:
The web content variant was rendering the columns and we can have more than one columns in a given web content attaching more event listeners than needed. Moving these event listeners to the Page component ensured we won't create duplicated event listeners.

Reply:
i see, thanks for the explanation!